src/components/card/card.tsx

```diff
@@ -0,0 +1,84 @@
+import { ICardProps } from "./types";
+import {
+    CardBody,
+    CardDragger,
+    CardIcon,
+    CardContent,
+    CardActions,
+    CardAction,
+    CardDetails,
+} from "./card.styles";
+import { BinIcon } from "@lifesg/react-icons/bin";
+import { CopyIcon } from "@lifesg/react-icons/copy";
+import { EnvelopeIcon } from "@lifesg/react-icons/envelope";
+import { EFieldType } from "src/schemas/types";
+import { Text } from "@lifesg/react-design-system/text";
+import { useState } from "react";
+
+export const Card = ({
+    field,
+    isFocused = false,
+    onClick,
+    onHover,
+    ...colProps
+}: ICardProps) => {
+    const [isHovered, setIsHovered] = useState(false);
+
+    const spansFull =
+        !colProps.desktopCols ||
+        (Array.isArray(colProps.desktopCols) &&
+            colProps.desktopCols.reduce((a, b) => a + b, 0) === 14);
+
+    const renderIcon = () => {
+        let icon: React.ReactNode;
+
+        switch (field.type) {
+            case EFieldType.EMAIL:
+                icon = <EnvelopeIcon />;
+                break;
+            default:
+                break;
+        }
+
+        return <CardIcon>{icon}</CardIcon>;
+    };
+
+    const handleHover = () => {
+        setIsHovered(!isHovered);
+        onHover?.();
+    };
+
+    return (
+        <CardContent
+            {...colProps}
+            $isFocused={isFocused}
+            onClick={onClick}
+            onMouseEnter={handleHover}
+            onMouseLeave={() => setIsHovered(false)}
+        >
+            <CardBody>
+                {/* TODO: placeholder dragger icon */}
+                {(isFocused || isHovered) && <CardDragger />}
```

Comment:
can we have a simple animation of the icon and text moving to the right when the dragger appears?